src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= threshold) {
+            resize();
+        }
 
+        if (table[hashIndex(key)] == null) {
+            table[hashIndex(key)] = new Node<>(key, value, null);
+            size++;
+            return;
+        }
+
+        Node<K, V> temp = table[hashIndex(key)];
+        while (temp.next != null || Objects.equals(temp.key, key)) {
+            if (Objects.equals(key, temp.key)) {
+                temp.value = value;
+                return;
+            }
+            temp = temp.next;
+        }
+        temp.next = new Node<>(key, value, null);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> temp = table[hashIndex(key)];
+        while (temp != null) {
+            if (Objects.equals(key, temp.key)) {
+                return temp.value;
+            }
+            temp = temp.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hashIndex(K key) {
```

Comment:
```suggestion
    private int getIndex(K key) {
```
you don't hash index you hash key to calculate index

Reply:
Thanx for suggestions. =)